source/postprocess/fluid_velocity_statistics.cc

```diff
@@ -0,0 +1,176 @@
+                                               gravity * (solid_density - fluid_density);
+                    local_fluid_velocity_square_integral += ((fluid_velocity_values[q] * fluid_velocity_values[q]) *
+                                                             fe_values.JxW(q));
+                    local_max_fluid_velocity = std::max (std::sqrt(fluid_velocity_values[q]*fluid_velocity_values[q]),
+                                                         local_max_fluid_velocity);
+                  }
+              }
+          }
+
+      const double global_fluid_velocity_square_integral
+        = Utilities::MPI::sum (local_fluid_velocity_square_integral, this->get_mpi_communicator());
+      const double global_max_fluid_velocity
+        = Utilities::MPI::max (local_max_fluid_velocity, this->get_mpi_communicator());
+
+      const double fluid_v_rms = std::sqrt(global_fluid_velocity_square_integral) /
+                                 std::sqrt(this->get_volume());
+
+      const std::string units = (this->convert_output_to_years() == true) ? "m/year" : "m/s";
+      const double unit_scale_factor = (this->convert_output_to_years() == true) ? year_in_seconds : 1.0;
+      const std::string advection_method = output_melt_velocity ? "melt" : "Darcy";
+      const std::vector<std::string> column_names = {"RMS " + advection_method + " velocity (" + units + ")",
+                                                     "Max. " + advection_method + " velocity (" + units + ")"
+                                                    };
+
+      statistics.add_value (column_names[0],
+                            fluid_v_rms * unit_scale_factor);
+      statistics.add_value (column_names[1],
+                            global_max_fluid_velocity * unit_scale_factor);
+
+      // also make sure that the other columns filled by this object
+      // all show up with sufficient accuracy and in scientific notation
+      for (auto &column : column_names)
+        {
+          statistics.set_precision (column, 8);
+          statistics.set_scientific (column, true);
+        }
+
+      std::ostringstream output;
+      output.precision(3);
+      output << fluid_v_rms *unit_scale_factor
+             << ' ' << units << ", "
+             << global_max_fluid_velocity *unit_scale_factor
+             << ' ' << units;
+
+      return std::pair<std::string, std::string> ("RMS " + advection_method + ", max " + advection_method + " velocity:",
+                                                  output.str());
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    ASPECT_REGISTER_POSTPROCESSOR(FluidVelocityStatistics,
+                                  "fluid velocity statistics",
+                                  "A postprocessor that computes the root mean square and "
+                                  "maximum velocity in the computational domain.")
```

Comment:
```suggestion
                                  "maximum fluid velocity in the computational domain.")
```
And can you add the fact that this will be the melt or Darcy velocity depending on what is chosen for the porosity field?